### ISSUE TITLE
Generate properties from parent class in subclass

### ISSUE THREAD

**Issue by jeanfei** [api-platform/schema-generator#14]
Hi,  

Thanks for this package. 
I try to generate entities with this app/config/schema.yml configuration : 

```
rdfa:
  - https://raw.githubusercontent.com/schemaorg/schemaorg/sdo-gozer/data/schema.rdfa
annotationGenerators:
    - SchemaOrgModel\AnnotationGenerator\PhpDocAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\DoctrineOrmAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\ConstraintAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\DunglasApiAnnotationGenerator
namespaces:
  entity: AppBundle\Entity
debug: true
types:
  Thing:
    properties:
      name: ~
      description: ~
      url: ~
  CreativeWork: ~
  WebSite:
    properties:
      mainEntityOfPage:
        range: WebPage
        cardinality: (0..1)
  WebPage:
    properties:
      mainEntityOfPage:
        range: Organization
        cardinality: (0..1)
      mainEntity:
        range: WebSite
        cardinality: (0..1)
  Organization: ~
```

I generate the entities with the command : bin/schema generate-types src/ app/config/schema.yml
My problem is that the 2 properties mainEntityOfPage (which belong to Thing) and mainEntity (which belong to CreativeWork) were not generated in WebSite and WebPage entities. 
I try to add directly mainEntityOfPage or mainEntity in subclasses because in parent classes, the targetEntity depends on the entity. 

What is the best way to manage that ? 
Do I have to insert manually mainEntityOfPage mainEntity attributes, and getter / setter ? 

Thanks,
Jean-Fei

**Comment by dunglas** [comment]
The generator do not support inheritance like that, try the following, it should work:

```yaml
rdfa:
  - https://raw.githubusercontent.com/schemaorg/schemaorg/sdo-gozer/data/schema.rdfa
annotationGenerators:
    - SchemaOrgModel\AnnotationGenerator\PhpDocAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\DoctrineOrmAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\ConstraintAnnotationGenerator
    - SchemaOrgModel\AnnotationGenerator\DunglasApiAnnotationGenerator
namespaces:
  entity: AppBundle\Entity
debug: true
types:
  Thing:
    properties:
      name: ~
      description: ~
      url: ~
      mainEntityOfPage:
        range: Organization
        cardinality: (0..1)
      mainEntity:
        range: WebSite
        cardinality: (0..1)
  CreativeWork: ~
  WebSite:
    properties: []
  WebPage:
    properties: []
  Organization: ~
```

If you want to move those properties from `Thing` to a child, then you're right, you must edit the generated model manually.

**Comment by jeanfei** [comment]
Thanks for this answer.
Actually, I can't use your solution because I use the property "mainEntityOfPage" in two subclasses (WebSite / WebPage), with a different targetEntity (WebPage / Organization).

Do you think that the TypesGenerator class should also generate those properties (even properties from parents) ? or edit them manually is the best way to do this ?

**Comment by aureluster** [comment]
+1

**Comment by dunglas** [comment]
Closed by #20
